{{cookiecutter.project_slug}}/compose/local/django/Dockerfile

```diff
@@ -1,19 +1,21 @@
-FROM python:3.7-alpine
+FROM debian:buster-slim
```

Comment:
Using this image instead of `python:3.7-slim-buster` would make it much harder to upgrade to Python 3.8, isn't there a way to get the python image to work? 

The Python it's using (3.7.3) is 9 months, not very convenient if one wants to upgrade to the latest.

Reply:
I'm sure there is a way to make the `python:3.7-slim-buster` work. It's a matter of efficiency and choice.

That image built Python 3.7.6 from source and put it at `/usr/local/bin/python`. This Python version looks for Python packages in `/usr/local/lib/python3.7/site-packages`.

In our Dockerfile, we install the package `python3-dev` which also installs its dependency, `python3` package. This version of Python(3.7.3) lives at `/usr/bin/python3` and sees Python packages installed in `/usr/local/lib/python3.7/dist-packages` and `/usr/lib/python3/dist-packages`.

Running just `python` in the `python:3.7-slim-buster` image (expectedly) uses its own Python 3.7.6 version, which does not see any system-distributed Python packages you may want to install. In my case, these are `python3-numpy`, `python3-scipy` (which would be installed into `/usr/lib/python3/dist-packages`). If you don't need those, the `python:3.7-slim-buster` image works fine with the default requirements files, albeit the fact that there are two versions of Python on the image.

If you require, say, numpy in your project, you can still install its wheel files without building but some C libraries it uses (i.e. libblas3, libgfortran5, liblapack3) are not installed. It may still work, but I'm not sure (or whether it works as fast).

So I think this is down to personal preference. I am happy with the Debian-packaged version of Python and like to be able to use Debian-packaged Python libraries without having to worry about their dependencies like above. An extra bonus is that I save some space taken by the second version of Python on the image.

Reply:
Just another note to say that I've tried the `python:3.7-slim-buster` and it works (as in all tests passed). The image size is 857 MB though.

If one really wants to use the `python:3.7-slim-buster` image and also want a small image, I guess we could purge the whole system Python version and its related packages once everything is built and installed.

Reply:
Further to my previous message, I've managed to get the `python:3.7-slim-buster` without having to install the Debian-packaged Python version (see commit 4a71b0b below). The resulted image size is now slightly smaller, 738 MB.

Reply:
Thanks for all these detailed explanations, good to clarify what was going on. And good work on keeping the image size down. I want to take a closer look, I hope to get some time this month to review all the outstading pull requests.